src/util/element-object-generator.ts

```diff
@@ -41,4 +42,48 @@ export namespace ElementObjectGenerator {
                 return null;
         }
     };
+
+    export const duplicate = (
+        element: TElement,
+        existingInternalIds: string[],
+        existingIds: string[]
+    ) => {
+        let internalId: string;
+        while (!internalId || existingInternalIds.includes(internalId)) {
+            internalId = SimpleIdGenerator.generate();
+        }
+
+        const extractBaseId = (id: string) => {
+            const match = id.match(/^(.*?)(-copy(?:-\d+)?)?$/);
+            return match ? match[1] : id;
+        };
+
+        const generateNewId = (baseId: string, existingIds: string[]) => {
+            const regex = new RegExp(`^${baseId}-copy(?:-(\\d+))?$`);
+            let maxSuffix = 0;
+
+            existingIds.forEach((existingId) => {
+                const match = existingId?.match(regex);
+                if (match) {
+                    const suffix = match[1] ? parseInt(match[1], 10) : 1;
+                    if (suffix > maxSuffix) {
+                        maxSuffix = suffix;
+                    }
+                }
+            });
```

Comment:
map all suffixes and then find the max once with `Math.max()`
```suggestion
            const suffixes = existingIds.map((existingId) => {
                const match = existingId?.match(regex);
                if (match) {
                    return match[1] ? parseInt(match[1], 10) : 1;
                }
                return 0;
            });
            const maxSuffix = Math.max(0, ...suffixes);
```

Reply:
changes are made on the latest push